### PR TITLE
added setPushToken method to the module interface

### PR DESCRIPTION
**Pull request by gabrielperales**

### CODE REVIEW

**Comment by fbartho** [comment]
@gabrielperales -- is this working for you? -- We don't want our Zendesk chat integrated with push-notifications in our app at this time, so I don't know that I'll be able to test it myself. But the code looks fine!

**Comment by gabrielperales** [comment]
I'm working on it, I made it work, but it suddenly stopped working... so I'm still testing. I'll keep you updated :)

**Comment by fbartho** [comment]
Sounds good @gabrielperales -- I know how hard debugging push notifications is, so good luck! Just tag me whenever you want me to review/merge/release this.

**Comment by fbartho** [comment]
@gabrielperales -- I'm starting to think about a new release, do you think you'll have time to get back to this? (and/or what's the status here?)

**Comment by gabrielperales** [comment]
I ended not using that and I wrote the code in objective C like this :

```
- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken
{
  [ZDKChat registerPushToken:deviceToken];
}
```

I had to test it more to see if it was working right, but I had some issues and I went to the path I knew it was working. Right now I'm pretty busy to test it again, but I would like to do it.

File: ios/RNZendeskChatModule.m [hunk 1]
```diff
@@ -191,4 +191,8 @@ - (void) dismissChatUI {
 	[ZDKChat initializeWithAccountKey:zenDeskKey queue:dispatch_get_main_queue()];
 }
 
+RCT_EXPORT_METHOD(registerPushToken:(NSString *)token) {
+	[ZDKChat registerPushToken:[RCTConvert NSData:token]];
```
Comment (fbartho) [inline ios/RNZendeskChatModule.m:195]:
```suggestion
	dispatch_sync(dispatch_get_main_queue(), ^{
		[ZDKChat registerPushToken:[RCTConvert NSData:token]];
	});
```

File: android/src/main/java/com/taskrabbit/zendesk/RNZendeskChatModule.java [hunk 2]
```diff
@@ -275,4 +275,13 @@ public void startChat(ReadableMap options) {
             Log.e(TAG, "Could not load getCurrentActivity -- no UI can be displayed without it.");
         }
     }
+
+	@ReactMethod
+    public void registerPushToken(String token) {
```
Comment (fbartho) [inline android/src/main/java/com/taskrabbit/zendesk/RNZendeskChatModule.java:280]:
```suggestion
    public void registerPushToken(String token) {
```

File: android/src/main/java/com/taskrabbit/zendesk/RNZendeskChatModule.java [hunk 3]
```diff
@@ -275,4 +275,13 @@ public void startChat(ReadableMap options) {
...
+		PushNotificationsProvider pushProvider = Chat.INSTANCE.providers().pushNotificationProvider();
+
+		if (pushProvider != null) {
+			pushProvider.registerPushToken(token);
+		}
+    }
```
Comment (fbartho) [inline android/src/main/java/com/taskrabbit/zendesk/RNZendeskChatModule.java:286]:
```suggestion
    }
```

**Review by fbartho — APPROVED** [review]
I think this code is okay, so I'm going to conditionally move forwards with it. Hopefully somebody will report if it works or doesn't work for them!

**Comment by fbartho** [comment]
This code is available with other changes in https://www.npmjs.com/package/react-native-zendesk-chat/v/0.4.1-beta.4

Note: there was a missing import / a typo on the Android side, so please check out the Beta!

**Comment by breadadams** [comment]
@fbartho do you want us to fix the missing import and typo or have you handled this already?

**Pull request merged**